File: D_Balanced_Round.py
t=int(input())
for ___ in range(t):
    n,k=map(int,input().split())
    v=list(map(int,input().split()))
    v.sort()
    i=0
    ans=0
    while i<n:
        cnt=1
        while i<n-1 and v[i+1]-v[i]<=k:
            i+=1
            cnt+=1
        ans=max(ans,cnt)
        i+=1
    print(n-ans)    
    

File: D_Balanced_Round.cpp
#include <bits/stdc++.h>
#define ll long long int
using namespace std;
void solve()
{
    int n, k;
    cin >> n >> k;
    vector<int> v(n);
    for (auto &x : v)
        cin >> x;
    sort(v.begin(), v.end());
    int ans = INT_MIN;
    for (int i = 0; i < n;)
    {
        int cnt = 1;
        while (i < n - 1 and (v[i + 1] - v[i]) <= k)
            cnt++, i++;
        i++;
        ans = max(ans, cnt);
    }
    cout << n - ans << endl;
}
int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int t = 1;
    cin >> t;
    for (int cs = 1; cs <= t; cs++)
    {
        // cout << "Case #" << cs << ": ";
        solve();
    }
    return 0;
}